index.js

```diff
@@ -104,4 +104,7 @@ function decode (data) {
   return cborg.decode(data, decodeOptions)
 }
 
-export default codec({ name, code, encode, decode })
+const decoder = new Decoder(name, code, decode)
+const encoder = new Encoder(name, code, encode)
```

Comment:
I’d suggest following instead:

```js
export const {name, code, encode, decode, encoder, decoder} = codec({ name, code, encode, decode })
```

That is because this would result in `code` and `name` getting types for corresponding literals as opposed to generalized `number` and `string` they get here.

Reply:
annotating those at definition site is also another option.

Reply:
> ```js
> export const {name, code, encode, decode, encoder, decoder} = codec({ name, code, encode, decode })
> ```

I tried that. But it fails to build (`SyntaxError: Identifier 'name' has already been declared (112:14)`). I thought it's a JS error, but now I realize it's probably a problem of our toolchain. I get:

```js
$ npm run build

> @ipld/dag-cbor@0.0.0-dev build
> npm run build:js && npm run build:types


> @ipld/dag-cbor@0.0.0-dev build:js
> npm_config_yes=true npx ipjs@latest build --tests

parsing file:///home/vmx/src/pl/js-dag-cbor/index.js
file:///home/vmx/.npm/_npx/ef1ca190cb7ddd1d/node_modules/acorn/dist/acorn.mjs:3010
  var err = new SyntaxError(message);
            ^

SyntaxError: Identifier 'name' has already been declared (112:14)
    at Parser.pp$4.raise (file:///home/vmx/.npm/_npx/ef1ca190cb7ddd1d/node_modules/acorn/dist/acorn.mjs:3010:13)
    at Parser.pp$5.declareName (file:///home/vmx/.npm/_npx/ef1ca190cb7ddd1d/node_modules/acorn/dist/acorn.mjs:3084:26)
    at Parser.pp$2.checkLValSimple (file:///home/vmx/.npm/_npx/ef1ca190cb7ddd1d/node_modules/acorn/dist/acorn.mjs:1899:48)
    at Parser.pp$2.checkLValPattern (file:///home/vmx/.npm/_npx/ef1ca190cb7ddd1d/node_modules/acorn/dist/acorn.mjs:1941:10)
    at Parser.pp$2.checkLValInnerPattern (file:///home/vmx/.npm/_npx/ef1ca190cb7ddd1d/node_modules/acorn/dist/acorn.mjs:1963:10)
    at Parser.pp$2.checkLValInnerPattern (file:///home/vmx/.npm/_npx/ef1ca190cb7ddd1d/node_modules/acorn/dist/acorn.mjs:1951:10)
    at Parser.pp$2.checkLValPattern (file:///home/vmx/.npm/_npx/ef1ca190cb7ddd1d/node_modules/acorn/dist/acorn.mjs:1928:10)
    at Parser.pp$1.parseVarId (file:///home/vmx/.npm/_npx/ef1ca190cb7ddd1d/node_modules/acorn/dist/acorn.mjs:1256:8)
    at Parser.pp$1.parseVar (file:///home/vmx/.npm/_npx/ef1ca190cb7ddd1d/node_modules/acorn/dist/acorn.mjs:1238:10)
    at Parser.pp$1.parseVarStatement (file:///home/vmx/.npm/_npx/ef1ca190cb7ddd1d/node_modules/acorn/dist/acorn.mjs:1104:8) {
  pos: 2740,
  loc: Position { line: 112, column: 14 },
  raisedAt: 2787
}
```

So I guess annotating them is the quicker way.

Reply:
It is a JS thing, I tried with Node.js:

```console
$ node index.js
file:///home/vmx/src/pl/js-dag-cbor/index.js:112
export const {name, code, encode, decode, encoder, decoder} = codec({ name, code, encode, decode })
              ^

SyntaxError: Identifier 'name' has already been declared
    at Loader.moduleStrategy (node:internal/modules/esm/translators:147:18)
    at async link (node:internal/modules/esm/module_job:48:21)
```

Reply:
> I tried that. But it fails to build (`SyntaxError: Identifier 'name' has already been declared (112:14)`). I thought it's a JS error, but now I realize it's probably a problem of our toolchain.

Ah that makes sense because we define `const name` and `const code` so it rejects `const { name, code ...`

Reply:
I think @rvagg made a change to a `codec` function so that it returns `encoder` and `decoder` fields, with that following should work:

```js
export const { name, code, decoder, encoder } = codec({ name: 'dag-cbor', code: 0x71, encode, decode })
```